### ISSUE TITLE
Exception parsing arguments as Long

### ISSUE THREAD

**Issue by okorz001** [andimarek/graphql-java#93]
Parsing an query that contains an argument that is larger the Integer.MAX_INT causes an exception to be thrown. This happens before schema validation, so it doesn't matter what the schema is.

Query: `{foo(bar:2147483648)}`

Exception:
```
java.lang.NumberFormatException: For input string: "2147483648"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:587)
	at java.lang.Integer.parseInt(Integer.java:615)
	at graphql.parser.GraphqlAntlrToLanguage.getValue(GraphqlAntlrToLanguage.java:324)
	at graphql.parser.GraphqlAntlrToLanguage.visitArgument(GraphqlAntlrToLanguage.java:272)
	at graphql.parser.GraphqlAntlrToLanguage.visitArgument(GraphqlAntlrToLanguage.java:14)
	at graphql.parser.antlr.GraphqlParser$ArgumentContext.accept(GraphqlParser.java:937)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
	at graphql.parser.antlr.GraphqlBaseVisitor.visitArguments(GraphqlBaseVisitor.java:155)
	at graphql.parser.antlr.GraphqlParser$ArgumentsContext.accept(GraphqlParser.java:876)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
	at graphql.parser.antlr.GraphqlBaseVisitor.visitField(GraphqlBaseVisitor.java:133)
	at graphql.parser.GraphqlAntlrToLanguage.visitField(GraphqlAntlrToLanguage.java:200)
	at graphql.parser.GraphqlAntlrToLanguage.visitField(GraphqlAntlrToLanguage.java:14)
	at graphql.parser.antlr.GraphqlParser$FieldContext.accept(GraphqlParser.java:754)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
	at graphql.parser.antlr.GraphqlBaseVisitor.visitSelection(GraphqlBaseVisitor.java:122)
	at graphql.parser.antlr.GraphqlParser$SelectionContext.accept(GraphqlParser.java:679)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
	at graphql.parser.antlr.GraphqlBaseVisitor.visitSelectionSet(GraphqlBaseVisitor.java:111)
	at graphql.parser.GraphqlAntlrToLanguage.visitSelectionSet(GraphqlAntlrToLanguage.java:185)
	at graphql.parser.GraphqlAntlrToLanguage.visitSelectionSet(GraphqlAntlrToLanguage.java:14)
 at graphql.parser.antlr.GraphqlParser$SelectionSetContext.accept(GraphqlParser.java:614)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
	at graphql.parser.antlr.GraphqlBaseVisitor.visitOperationDefinition(GraphqlBaseVisitor.java:45)
	at graphql.parser.GraphqlAntlrToLanguage.visitOperationDefinition(GraphqlAntlrToLanguage.java:121)
	at graphql.parser.GraphqlAntlrToLanguage.visitOperationDefinition(GraphqlAntlrToLanguage.java:14)
	at graphql.parser.antlr.GraphqlParser$OperationDefinitionContext.accept(GraphqlParser.java:271)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
	at graphql.parser.antlr.GraphqlBaseVisitor.visitDefinition(GraphqlBaseVisitor.java:34)
	at graphql.parser.antlr.GraphqlParser$DefinitionContext.accept(GraphqlParser.java:198)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
	at graphql.parser.antlr.GraphqlBaseVisitor.visitDocument(GraphqlBaseVisitor.java:23)
	at graphql.parser.GraphqlAntlrToLanguage.visitDocument(GraphqlAntlrToLanguage.java:102)
	at graphql.parser.Parser.parseDocument(Parser.java:32)
	at graphql.GraphQL.execute(GraphQL.java:64)
	at graphql.GraphQL.execute(GraphQL.java:55)
	at graphql.GraphQL.execute(GraphQL.java:47)
	at graphql.GraphQL.execute(GraphQL.java:43)
```

Any value consisting only of digits becomes IntValue:
https://github.com/andimarek/graphql-java/blob/master/src/main/grammar/Graphql.g4#L116

IntValue is cast into Integer:
https://github.com/andimarek/graphql-java/blob/master/src/main/java/graphql/parser/GraphqlAntlrToLanguage.java#L323-L326

Perhaps it could parse into Long if Integer parse fails, and parse into BigInteger if Long parse fails.

**Comment by okorz001** [comment]
Alternatively, I just realized FloatValue is always parsed into a BigDecimal, so I suppose it could just unconditionally parse IntValue into BigInteger.

**Comment by andimarek** [comment]
The exception is not pretty, but I think it is correct not to parse such big values:

According to the spec Integers which are too big should raise errors:
http://facebook.github.io/graphql/#sec-Int

If you wanna use Long Values, you can write them down as Strings.

I hope this helps.

**Comment by okorz001** [comment]
Yes, for GraphQLInt values, I agree that there must be an exception thrown somewhere.

I am currently working around this by using String as input and Long as output but this asymmetry is inconvenient to clients. It is possible to use GraphQLLong as output without any issues.

I think we should defer the error from token parsing and wait until type deserialization. Instead of retrieving the value unconditionally here, we could error out if the value is out of range.
GraphQLInt: https://github.com/andimarek/graphql-java/blob/master/src/main/java/graphql/Scalars.java#L31-L36
GraphQLLong: https://github.com/andimarek/graphql-java/blob/master/src/main/java/graphql/Scalars.java#L58-L66

**Comment by okorz001** [comment]
I am working on proof of this, but I am still having some test failures that I must investigate.
https://github.com/andimarek/graphql-java/compare/master...okorz001:allow-longs

**Comment by okorz001** [comment]
I have opened #94. I would like to add negative tests to prove GraphQLInt will still throw for values greater than MAX_INT, but I am unfamiliar with both Groovy and Spock.

**Comment by andimarek** [comment]
This is solved by #94 and https://github.com/andimarek/graphql-java/commit/aabed3ae8f08550740f58b1f83c727ed24630437
